File: Code/Disney/disney.xBandController/src/windows/WDW.NGE.Support.Models/xBMS/Shipment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SimpleMvvmToolkit;

namespace WDW.NGE.Support.Models.xBMS
{
    public class Shipment : ModelBase<Shipment>
    {
        private Address address;
        private String method;
        private String carrier;
        private String carrierLink;
        private String trackingNumber;
        private String shippingDate;

        public Address Address
        {
            get { return this.address; }
            set
            {
                this.address = value;
                NotifyPropertyChanged(m => m.Address);

            }
        }

        public String Method
        {
            get { return this.method; }
            set
            {
                this.method = value;
                NotifyPropertyChanged(m => m.Method);

            }
        }

        public String Carrier
        {
            get { return this.carrier; }
            set
            {
                this.carrier = value;
                NotifyPropertyChanged(m => m.Carrier);

            }
        }

        public String CarrierLink
        {
            get { return this.carrierLink; }
            set
            {
                this.carrierLink = value;
                NotifyPropertyChanged(m => m.CarrierLink);

            }
        }

        public String TrackingNumber
        {
            get { return this.trackingNumber; }
            set
            {
                this.trackingNumber = value;
                NotifyPropertyChanged(m => m.TrackingNumber);

            }
        }

        public String ShippingDate
        {
            get { return this.shippingDate; }
            set
            {
                this.shippingDate = value;
                NotifyPropertyChanged(m => m.ShippingDate);

            }
        }
    }
}

File: Code/Disney/disney.xBandController/src/linux/JmsLib/src/main/java/com/disney/xband/jms/lib/entity/xbms/Shipment.java
package com.disney.xband.jms.lib.entity.xbms;

import javax.xml.bind.annotation.XmlElement;

/*
"method": "PRIMARY_GUEST_ADDRESS_BEST",
"carrier" : "UPS",
"carrierLink" : "http://www.ups.com/WebTracking/track",
"shippingDate" : "2010-08-27T04:00:00Z",
"trackingNumber" : 66,
"address" : {
 */

public class Shipment
{
	private String method;
	private String carrier;
	private String carrierLink;
	private String shippingDate;
	private String trackingNumber;
	private Address address;

	@XmlElement(name="method")
	public String getMethod()
	{
		return this.method;
	}

	public void setMethod(String method)
	{
		this.method = method;
	}
	
	@XmlElement(name="carrier")
	public String getCarrier()
	{
		return this.carrier;
	}

	public void setCarrier(String carrier)
	{
		this.carrier = carrier;
	}
	
	@XmlElement(name="carrierLink")
	public String getCarrierLink()
	{
		return this.carrierLink;
	}

	public void setCarrierLink(String carrierLink)
	{
		this.carrierLink = carrierLink;
	}
	
	@XmlElement(name="shippingDate")
	public String getShippingDate()
	{
		return this.shippingDate;
	}

	public void setShippingDate(String shippingDate)
	{
		this.shippingDate = shippingDate;
	}
	@XmlElement(name="trackingNumber")
	public String getTrackingNumber()
	{
		return this.trackingNumber;
	}

	public void setTrackingNumber(String trackingNumber)
	{
		this.trackingNumber = trackingNumber;
	}
	
	@XmlElement(name="address")
	public Address getAddress()
	{
		return this.address;
	}

	public void setAddress(Address address)
	{
		this.address = address;
	}
}
